### ISSUE TITLE
Discussion: Add lazy attribute helper (emberjs style bindAttr templateHelper)

### ISSUE THREAD

**Issue by g00fy-** [marionettejs/backbone.marionette#492]
Since I don't really know if Marionette is a 'building foundation' or fully-featured framework I allowed myself to add this proposal, to push new features to Marionette.
## TL;DR : proof of concept fiddle:

http://jsfiddle.net/g00fy/7VRJw/

Let's consider a model that has multiple relations to other models but itself contains only id of the related model:

``` javascript
// a car model json
{
  id:1,
  make:'toyota',
  owner_id:2
}  
```

Imagine that we would like to show in our view owners name. 

Currenty what we can do is to:
1. Render the view without this information (since extra request may take some time & we don't want a user to wait)
2. Fetch Owner & set it on Car  `car.set({owner:owner})`
3. Render the view again

We could also use `ui` or declare extra events - but this gets messy when you have to deal with multiple relations & many models;

What if we could have a `lazy` attribute, that would update itself when this attribute is available.

example use would be:

``` html
<h1>Owner : <%= lazyAttr('owner.name') %></h1>
```
#### How it works:

**tl;dr** Same way like emberjs bindings - using two `script` tags html & referring to `textNode` between them. 
1. When `lazyAttr` helper gets invocated, it listens to `change` on the model, returns html with two `script` tags & tries to return the value (if already present).
2. On `change`, callback function replaces `textNode.textContext` with new value

if you want, you can checkout the ugly draft code https://gist.github.com/g00fy-/5027167

I have developed a proof of concept & am awaiting suggestions or comments.

This of course would only work for text nodes (not inside tags), but it is a lot faster than re-rendering a list of 100 models;

AFAIK - this would be also template engine agnostic (at least for those templates that support helpers).
